src/demo/OceanObject.js

```diff
@@ -59,12 +58,13 @@ export const loadAllFishPartImages = () => {
   );
 };
 
-export const generateOceanObject = (allowedClasses, id, dataSet = null) => {
+export const generateOceanObject = (
+  allowedClasses,
+  id,
+  allowedComponents = fishData
+) => {
   const idx = Math.floor(Math.random() * allowedClasses.length);
-  const newOceanObject = new allowedClasses[idx](
-    id,
-    filterFishComponents(fishData, dataSet)
-  );
+  const newOceanObject = new allowedClasses[idx](id, allowedComponents);
```

Comment:
hmm I'm curious how this will work when we have other types of ocean objects?

Reply:
Can you update the method definition on line 73 to reflect this change?

Reply:
ah, good point -- i'll refactor to make this work for other types of ocean objects. i think line 73 will stay the same for now because only `FishOceanObject` will have components supplied to its constructor